src/nrdp/NrdpSSLNetworkPipe.ts

```diff
@@ -170,11 +170,12 @@ class NrdpSSLNetworkPipe extends NetworkPipe {
         }
     }
 
-    read(buf: IDataBuffer, offset: number, length: number, noStash?: boolean): number {
-        if (!noStash) {
+    read(buf: IDataBuffer, offset: number, length: number): number {
+        if (this.hasStash()) {
```

Comment:
I don't think there exists a universe where you wouldn't just implicitly read stashed data first.